src/module-elasticsuite-virtual-category/view/adminhtml/web/js/component/catalog/category/form/assign-products.js

```diff
@@ -44,8 +44,8 @@ define([
                 if (element.positionElement) {
                     element.positionElement.disabled = false;
                     categoryProducts.set(element.value, element.positionElement.value);
-                } else if (element.value && element.id) {
-                    // Override is here. Add the product to categoryProducts even if it does not have position field
+                } else if (parseInt(element.value, 10) > 0) {
```

Comment:
Please change to `if (!isNan(parseInt(element.value, 10))) {`, I'm more confident in the behavior of isNan than in the `>` comparison.